packages/server/src/worker/processJobSyncAlgolia.ts

```diff
@@ -29,36 +29,59 @@ export async function processJobSyncAlgolia(
   job.updateProgress(25);
 
   let wasUpdated = false;
-  Resource.getAll()
+  let errorMessage = null;
+  Resource.find({})
```

Comment:
So for now this is great. Lets get it shipped.

Now the next level stuff:

The reason `Resource.getAll` existed but then was removed (whoops.. left the type defs.. #366 ) was because for dev it's fine, but as the dataset grows putting all the results in memory becomes unrealistic. So next level would be to break this up in batches.. fetch 10 (or 20 or 100) resources, sync them to algolia, then do the next batch. This has a few more advantages: we can provide a more realistic arg to `updateProgress` since we can call [`estimatedDocumentCount`](https://docs.mongodb.com/manual/reference/method/db.collection.estimatedDocumentCount/) on the collection at the start, then we'll know the percentage of the way through the records we are for each batch and we'll send a progress update. This also will have implications on the network side when we make the request to algolia. Smaller requests are more likely to succeed for one, and also there's probably a limit on how much data we can send to algolia at once (https://www.algolia.com/doc/guides/sending-and-managing-data/send-and-update-your-data/#batching).

On the flip side this is more complicated. And we have to worry about more failure modes. What happens if batch #4 of 27 fails when we try to push it algolia. Does our job die? Do the rest get pushed but those are missing in the index? Do we retry that failure? How quickly? How many times before we give up? Do we need to retry the whole job?

So for our tiny dataset, this is fine, but you'll def run into the need for batching sooner or later if you haven't already.

Reply:
Also.. this function is `async` already. I'd probably want to `await` these promises